query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/select/CanAddSelector.java

```diff
@@ -0,0 +1,710 @@
+   * selector(QueryBuilderDsl.getDifference(left, right))}.
+   *
+   * @see QueryBuilderDsl#getDifference(Selector, Selector)
+   */
+  default Select difference(Selector left, Selector right) {
+    return selector(QueryBuilderDsl.getDifference(left, right));
+  }
+
+  /**
+   * Selects the product of two arguments, as in {@code SELECT col1 * col2}.
+   *
+   * <p>This is available in Cassandra 4 and above.
+   *
+   * <p>This is a shortcut for {@link #selector(Selector) selector(QueryBuilderDsl.getProduct(left,
+   * right))}.
+   *
+   * <p>The arguments will be parenthesized if they are instances of {@link QueryBuilderDsl#getSum}
+   * or {@link QueryBuilderDsl#getDifference}. If they are raw selectors, you might have to
+   * parenthesize them yourself.
+   *
+   * @see QueryBuilderDsl#getProduct(Selector, Selector)
+   */
+  default Select product(Selector left, Selector right) {
+    return selector(QueryBuilderDsl.getProduct(left, right));
+  }
+
+  /**
+   * Selects the quotient of two arguments, as in {@code SELECT col1 / col2}.
+   *
+   * <p>This is available in Cassandra 4 and above.
+   *
+   * <p>This is a shortcut for {@link #selector(Selector) selector(QueryBuilderDsl.getQuotient(left,
+   * right))}.
+   *
+   * <p>The arguments will be parenthesized if they are instances of {@link QueryBuilderDsl#getSum}
+   * or {@link QueryBuilderDsl#getDifference}. If they are raw selectors, you might have to
+   * parenthesize them yourself.
+   *
+   * @see QueryBuilderDsl#getQuotient(Selector, Selector)
+   */
+  default Select quotient(Selector left, Selector right) {
+    return selector(QueryBuilderDsl.getQuotient(left, right));
+  }
+
+  /**
+   * Selects the remainder of two arguments, as in {@code SELECT col1 % col2}.
+   *
+   * <p>This is available in Cassandra 4 and above.
+   *
+   * <p>This is a shortcut for {@link #selector(Selector)
+   * selector(QueryBuilderDsl.getRemainder(left, right))}.
+   *
+   * <p>The arguments will be parenthesized if they are instances of {@link QueryBuilderDsl#getSum}
+   * or {@link QueryBuilderDsl#getDifference}. If they are raw selectors, you might have to
+   * parenthesize them yourself.
+   *
+   * @see QueryBuilderDsl#getRemainder(Selector, Selector)
+   */
+  default Select remainder(Selector left, Selector right) {
+    return selector(QueryBuilderDsl.getRemainder(left, right));
+  }
+
+  /**
+   * Selects the opposite of an argument, as in {@code SELECT -col1}.
+   *
+   * <p>This is available in Cassandra 4 and above.
+   *
+   * <p>This is a shortcut for {@link #selector(Selector)
+   * selector(QueryBuilderDsl.getOpposite(argument))}.
+   *
+   * <p>The argument will be parenthesized if it is an instance of {@link QueryBuilderDsl#getSum} or
+   * {@link QueryBuilderDsl#getDifference}. If it is a raw selector, you might have to parenthesize
+   * it yourself.
+   *
+   * @see QueryBuilderDsl#getOpposite(Selector)
+   */
+  default Select opposite(Selector argument) {
```

Comment:
Hmm I see, you are trying to use substantives that denote the resulting state rather than the modifying action, such as "opposite", "difference", "sum", "quotient"... I agree that at least that's 100% consistent, but wouldn't users be more accustomed to traditional arithmetic operation names such as "negation", "subtraction", "addition", "division"...? (totally open question)

Reply:
Yes, I chose the formal names because they're the most precise:

* I believe "negation" is more of a boolean logic term
* technically, `%` is a division as well. Integer division produces two results, the quotient and the remainder.

But maybe I'm being too much of a nerd here, and it's true that when looking for division in auto-complete, "quotient" is probably not what first comes to mind. Maybe we can use the operation names, and keep `remainder()` even if it's not consistent with the others.

Reply:
To be clear, I find your initial set of names very precise, but as you pointed out, my fear is that people will struggle looking for them in their IDE. Something else came to me since: How about we stay consistent with the `BigDecimal` class? It mostly uses verbs (multiply, divide, add, subtract), and also "remainder" and "negate".

Reply:
👍 , it's better to be consistent with well-known platform conventions.